[16.0][FIX] product_tax_multicompany_default: Improve test coverage

Shide opened the pull request:
Testing `product.set_multicompany_taxes()` on `product_tax_multicompany_default ` module

Cherry pick of https://github.com/OCA/multi-company/pull/447/commits/9645c0c54c5cc45649d274fde607155e2cb998c9

MT-2121 @moduon @rafaelbn @carlosdauden @yajo

pedrobaeza commented on product_tax_multicompany_default/tests/test_product_tax_multicompany.py:
```diff
@@ -17,27 +19,49 @@ def setUpClass(cls):
+        group_account_user = cls.env.ref("account.group_account_user")
+        group_multi_company = cls.env.ref("base.group_multi_company")
         ResUsers = cls.env["res.users"]
         cls.user_1 = ResUsers.create(
```
Please use the method `test_new_user` for simplifying code:

https://github.com/odoo/odoo/blob/08859ca7939454303e34dc39c831ac8f75773c3d/odoo/tests/common.py#L124

Example: https://github.com/odoo/odoo/blob/08859ca7939454303e34dc39c831ac8f75773c3d/addons/web/tests/test_ir_model.py#L37

Shide replied:
Done

pedrobaeza commented on product_tax_multicompany_default/tests/test_product_tax_multicompany.py:
```diff
@@ -131,3 +167,75 @@ def test_tax_not_default_set_match(self):
         self.assertIn(self.tax_10_cc2, product.taxes_id)
         self.assertIn(self.tax_10_sc2, product.supplier_taxes_id)
+
+    def test_set_multicompany_taxes(self):
```
Why not decorating this with `@users` for testing with such user/s?

Shide replied:
Done

pedrobaeza reviewed (changes requested):
Please put the commit message properly

Shide committed:
[FIX] product_tax_multicompany_default: Improve test coverage

- Refactored some tests to use new_test_user and @users
- Added a new test to cover the workflow of the product Form

pedrobaeza reviewed (approved):
/ocabot merge patch
